includes/Commerce/Orders.php

```diff
@@ -96,10 +99,190 @@ public function create_local_order( Order $remote_order ) {
+		}
+
+		$matching_shipping_order_item->set_total( $selected_shipping_option['price']['amount'] );
+		$matching_shipping_order_item->set_taxes( [
+			'total' => [ $selected_shipping_option['calculated_tax']['amount'] ],
+		] );
+		$matching_shipping_order_item->save();
+
+		// add tax item
+		$matching_tax_order_item = false;
+
+		// check if the local order already has a tax item item
+		if ( ! empty( $tax_order_items = $local_order->get_items( 'tax' ) ) ) {
+			$matching_tax_order_item = current( $tax_order_items );
+		}
+
+		if ( empty( $matching_tax_order_item ) ) {
+
+			$matching_tax_order_item = new \WC_Order_Item_Tax();
+			$local_order->add_item( $matching_tax_order_item );
+		}
+
+		$matching_tax_order_item->set_tax_total( $total_items_tax );
+		$matching_tax_order_item->set_shipping_tax_total( $selected_shipping_option['calculated_tax']['amount'] );
+		$matching_tax_order_item->save();
+
+		$local_order->set_shipping_total( $selected_shipping_option['price']['amount'] );
+		$local_order->set_shipping_tax( $selected_shipping_option['calculated_tax']['amount'] );
+
+		// update information from shipping_address
+		$shipping_address = $remote_order->get_shipping_address();
+
+		if ( ! empty( $shipping_address['name'] ) ) {
+
+			if ( strpos( $shipping_address['name'], ' ' ) !== false ) {
+
+				list( $first_name, $last_name ) = explode( ' ', $shipping_address['name'], 2 );
+				$local_order->set_shipping_first_name( $first_name );
+				$local_order->set_shipping_last_name( $last_name );
+
+			} else {
+
+				$local_order->set_shipping_last_name( $shipping_address['name'] );
+			}
+		}
+
+		if ( ! empty( $shipping_address['street1'] ) ) {
+			$local_order->set_shipping_address_1( $shipping_address['street1'] );
+		}
+
+		if ( ! empty( $shipping_address['street2'] ) ) {
+			$local_order->set_shipping_address_2( $shipping_address['street2'] );
+		}
+
+		if ( ! empty( $shipping_address['city'] ) ) {
+			$local_order->set_shipping_city( $shipping_address['city'] );
+		}
+
+		if ( ! empty( $shipping_address['state'] ) ) {
+			$local_order->set_shipping_state( $shipping_address['state'] );
+		}
+
+		if ( ! empty( $shipping_address['postal_code'] ) ) {
+			$local_order->set_shipping_postcode( $shipping_address['postal_code'] );
+		}
+
+		if ( ! empty( $shipping_address['country'] ) ) {
+			$local_order->set_shipping_country( $shipping_address['country'] );
+		}
+
+		// update information from estimated_payment_details
+		$estimated_payment_details = $remote_order->get_estimated_payment_details();
+
+		// we do not use subtotal values from the API because WC calculates them on the fly based on the items
+		$local_order->set_total( $estimated_payment_details['total_amount']['amount'] );
+		$local_order->set_currency( $estimated_payment_details['total_amount']['currency'] );
+
+		// update information from buyer_details
+		$buyer_details = $remote_order->get_buyer_details();
+
+		if ( strpos( $buyer_details['name'], ' ' ) !== false ) {
```

Comment:
It looks like the buyer details are missing if the order is still `FB_PROCESSING`, so this throws a notice in that case.

Reply:
Fixed in https://github.com/facebookincubator/facebook-for-woocommerce/pull/1515/commits/7b725748e1aac89a8cdb1347f0d604f28f7b0489.

Reply:
I'm afraid I still see notices for this case - maybe you'll need to check the individual parameters?

```
PHP Notice:  Undefined index: name in includes/Commerce/Orders.php on line 264
PHP Notice:  Undefined index: name in includes/Commerce/Orders.php on line 272
PHP Notice:  Undefined index: email in includes/Commerce/Orders.php on line 275
```

Reply:
This is weird, can you send me the API response from the logs for the `FB_PROCESSING` order?

Reply:
sure! Looks like `email_remarketing_option` still returns regardless

```json
{
  "data": [
    {
      "id": "*********",
      "order_status": {
        "state": "FB_PROCESSING"
      },
      "created": "2020-09-04T18:34:57+00:00",
      "last_updated": "2020-09-04T18:35:08+00:00",
      "items": {
        "data": [
          {
            "id": "*********",
            "retailer_id": "*********",
            "product_id": "*********",
            "quantity": 3,
            "price_per_unit": {
              "amount": "1.00",
              "currency": "USD"
            },
            "tax_details": {
              "estimated_tax": {
                "amount": "0.29",
                "currency": "USD"
              },
              "captured_tax": {
                "total_tax": {
                  "amount": "0.00",
                  "currency": "USD"
                }
              }
            }
          }
        ],
        "paging": {
          "cursors": {
            "before": "*********",
            "after": "*********"
          }
        }
      },
      "ship_by_date": "2020-09-10",
      "channel": "facebook",
      "selected_shipping_option": {
        "name": "STANDARD (3 - 5 Business days)",
        "price": {
          "amount": "1.00",
          "currency": "USD"
        },
        "calculated_tax": {
          "amount": "0.09",
          "currency": "USD"
        },
        "estimated_shipping_time": {
          "min_days": 3,
          "max_days": 5
        }
      },
      "estimated_payment_details": {
        "subtotal": {
          "items": {
            "amount": "3.00",
            "currency": "USD"
          },
          "shipping": {
            "amount": "1.00",
            "currency": "USD"
          }
        },
        "tax": {
          "amount": "0.38",
          "currency": "USD"
        },
        "total_amount": {
          "amount": "4.38",
          "currency": "USD"
        },
        "tax_remitted": true
      },
      "buyer_details": {
        "email_remarketing_option": false
      }
    }
  ],
  "paging": {
    "cursors": {
      "before": "*********",
      "after": "*********"
    }
  }
}
```

Reply:
Yikes, could they be weirder? 🤷‍♀️ 
I'll tweak the ifs...